packages/ses/src-xs/index.js

```diff
@@ -0,0 +1,463 @@
+    };
+  }
+  // Legacy support for record descriptors.
+  if (descriptor.record !== undefined) {
+    if (
+      descriptor.specifier === specifier ||
+      descriptor.specifier === undefined
+    ) {
+      return {
+        source: adaptModuleSource(descriptor.record),
+        specifier,
+        importMeta: descriptor.importMeta,
+      };
+    } else {
+      if (compartment === undefined) {
+        throw new TypeError(
+          'Cannot construct forward reference module descriptor in module map',
+        );
+      }
+      const { descriptors } = weakmapGet(privateFields, compartment);
+      mapSet(descriptors, descriptor.specifier, {
+        compartment,
+        namespace: specifier,
+      });
+      return {
+        source: adaptModuleSource(descriptor.record),
+        specifier: descriptor.specifier,
+        importMeta: descriptor.importMeta,
+      };
+    }
+  }
+  if (descriptor.specifier !== undefined) {
+    return {
+      namespace: descriptor.specifier,
+      compartment: descriptor.compartment,
+    };
+  }
+  // Legacy support for a source in the place of a descriptor.
+  return { source: adaptModuleSource(descriptor) };
+};
+
+const commons = {
+  Map,
+  TypeError,
+  adaptModuleDescriptor,
+  arrayMap,
+  compartmentOptions,
+  defineProperty,
+  entries,
+  fromEntries,
+  mapDelete,
+  mapGet,
+  nativeEvaluate,
+  nativeGetGlobalThis,
+  privateFields,
+  shimGetGlobalThis,
+  uncurryThis,
+  weakmapSet,
+};
+
+// This machinery allows us to replace the native Compartment with an adapter
+// in the start compartment and any child compartment that the adapter begets.
+const compartmentShim = `(
+  compartmentShim,
+  commons,
+  NativeCompartment,
+  ShimCompartment,
+  maybeHarden,
+) => {
+  const {
+    Map,
+    TypeError,
+    adaptModuleDescriptor,
+    arrayMap,
+    compartmentOptions,
+    defineProperty,
+    entries,
+    fromEntries,
+    mapDelete,
+    mapGet,
+    nativeEvaluate,
+    nativeGetGlobalThis,
+    privateFields,
+    shimGetGlobalThis,
+    uncurryThis,
+    weakmapSet,
+  } = commons;
+
+  function Compartment(...args) {
+    const options = compartmentOptions(...args);
+
+    const {
+      name = undefined,
+      globals = {},
+      transforms = [],
+      resolveHook = () => {
+        throw new TypeError('Compartment requires a resolveHook');
+      },
+      loadHook = undefined,
+      loadNowHook = undefined,
+      importHook = loadHook,
+      importNowHook = loadNowHook,
+      moduleMapHook = () => {},
+      __native__: delegateNative = false,
+      __noNamespaceBox__: noNamespaceBox = false,
+    } = options;
+
+    const modules = delegateNative
+      ? fromEntries(
+          arrayMap(
+            entries(options.modules ?? {}),
+            ([specifier, descriptor]) => [
+              specifier,
+              adaptModuleDescriptor(descriptor, specifier, undefined),
+            ],
+          ),
+        )
+      : {};
```

Comment:
Add comment explaining that this `module` is only used for `nativeOptions` and the shim options pass `options.modules` thru unaltered. Arguably, the `{}` at the end could be anything, but `{}` satisfies the type.